### PR TITLE
Lpal 457 add infra and build details for perfplat worker

### PR DESCRIPTION
**Pull request by williamfalconeruk**
## Purpose

To add the infra and pipeline for the new performance-platform module in Make. We have based this on the infra in the Use an LPA team, but made it environment level as this means we do not share the lambda across ephemeral environments.

initially switched off by using a feature flag: `performance_platform_enabled` in `terraform/environment/terraform.tfvars.json`. 
We will commit a switch on when ready. 
**NOTE**: This is enabled for Development only at this time.

Fixes LPAL-457

## Approach

- create lambda infra and module
- add sqs infra
- update dockerfiles to have a CI element
- add CircleCI jobs and commands, and update workflows
- terraform feature flag
- tfsec scan

## Learning

See https://github.com/ministryofjustice/opg-use-an-lpa/pull/782 for examples used in this.

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [X] I have added mandatory tags to terraformed resources, where possible
* [ ] If I added a package.json or composer.json, I also made sure this is included in the script in `.github/workflows/dependabot-update.yml`
* [ ] The product team have tested these changes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -205,6 +216,24 @@ path_to_live: &path_to_live
         unit_test: false
         filters: { branches: { only: [master] } }
 
+    - build_containers_and_push_to_ecr/docker_build_lambda_image:
+        name: perfplat-lambda-docker-build-preproduction
+        account_id: "987830934591"
```
Comment (townxelliot) [inline .circleci/config.yml:221]: Should this be a configuration variable?

**Review by townxelliot — APPROVED** [review]
Looks OK to me so far as my limited knowledge of Terraform and AWS goes.

**Comment by townxelliot** [comment]
Is there any way we can manually test the dev SQS and lambda once it's up, just to ensure the event source is working correctly.

File: .circleci/config.yml (hunk 1 above)
Comment (williamfalconeruk) [reply .circleci/config.yml:221]: ideally, our counterparts on the Use team also do this in their CI config however. might be less of an issue on the migration to Github Actions.

**Comment by williamfalconeruk** [comment]
> Is there any way we can manually test the dev SQS and lambda once it's up, just to ensure the event source is working correctly?

I'll look into this for you now.

**Pull request merged**